assets/app/view/game/buy_sell_shares.rb

```diff
@@ -16,13 +16,13 @@ def render
         @step = @game.round.active_step
         @current_entity = @step.current_entity
 
-        @ipo_shares = @corporation.ipo_shares.group_by(&:percent).values
+        @ipo_shares = @corporation.ipo_shares.select(&:buyable).group_by(&:percent).values
```

Comment:
can you check that this doesn't break other games? shared view code is very dangerous these days

Reply:
One of the first things that the farther down loops do is check if each share is buyable with one of the `can_buy` methods. I can do further testing if you want of course, but I think we're good here. The main thing it does is that if there are unbuyable shares on the top of the stack with buyable shares beneath we didn't show the button. I don't know of a game that relies on that (actually checking the `buyable` field, not just buyable due to limits etc).